analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -18,14 +18,72 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
 
 public abstract class ConditionEvaluatesToConstantBase : SymbolicRuleCheck
 {
-    protected const string DiagnosticIdCodeSmell = "S2589"; // Code smell
-    protected const string DiagnosticIdBug = "S2583"; // Bug
+    protected const string DiagnosticId2583 = "S2583"; // Bug
+    protected const string DiagnosticId2589 = "S2589"; // Code smell
 
     protected const string MessageFormat = "{0}";
-    protected const string MessageFormatBool = "Change this condition so that it does not always evaluate to '{0}'.";
-    protected const string MessageNull = "Change this expression which always evaluates to 'null'.";
+    private const string MessageBool = "Change this condition so that it does not always evaluate to {1}";
```

Comment:
```suggestion
    private const string MessageBool = "Change this condition so that it does not always evaluate to '{1}'";
```

Reply:
You did not apply this. But I just realized that you have it on the caller side. I would move it here instead.